src/ProjectNeon/Assets/Features/Effects/RepeatOnPrimaryResource.cs

```diff
@@ -0,0 +1,23 @@
+﻿using UnityEngine;
+using System.Collections;
+
+/**
+ * Apply an effect while there is enough Primary Resource
+ */
+public class RepeatOnPrimaryResource : Effect
+{
+    private Effect _effect;
+
+    public RepeatOnPrimaryResource(Effect effect)
+    {
+        _effect = effect;
+    }
+
+    void Effect.Apply(Member source, Target target)
+    {
+        while (source.State[source.State.ResourceTypes[0]]  > 0)
```

Comment:
Having more than 0 doesn't guarantee that the effect can be afforded. Should be ensuring that it has enough to pay for the recurring effect.

Also, this double-indexing means that we need a more convenient getter on `MemberState`.